api_app/ingestors_manager/ingestors/malware_bazaar.py

```diff
@@ -0,0 +1,158 @@
+import io
+import logging
+import time
+from datetime import datetime
+from typing import Any, Iterable
+from unittest.mock import patch
+
+import pyzipper
+import requests
+
+from api_app.ingestors_manager.classes import Ingestor
+from api_app.ingestors_manager.exceptions import IngestorRunException
+from tests.mock_utils import MockUpResponse, if_mock_connections
+
+logger = logging.getLogger(__name__)
+
+
+class MalwareBazaar(Ingestor):
+    # API endpoint
+    url: str
+    # Download samples that are up to X hours old
+    hours: int
+    # Download samples from chosen signatures (aka malware families)
+    signatures: str
+
+    @classmethod
+    def update(cls) -> bool:
+        pass
+
+    # retrieve information about the given signature
+    def get_signature_information(self, signature):
+        result = requests.post(
+            self.url,
+            data={"query": "get_siginfo", "signature": signature, "limit": 100},
+            timeout=30,
+        )
+        result.raise_for_status()
+        content = result.json()
+        logger.info(f"Malware bazaar data for signature {signature} is {content}")
+        if content["query_status"] != "ok":
+            raise IngestorRunException(
+                f"Query status is invalid: {content['query_status']}"
+            )
+        if not isinstance(content["data"], list):
+            raise IngestorRunException(f"Content {content} not expected")
+        return content["data"]
+
+    # extract file hashes per signature
+    def get_recent_samples(self):
+        hashes = set()
+        now = datetime.now()
```

Comment:
django.utils.timezone.now instead of datetime. Just to be coherent with the rest of the application